analyses/copy_number_consensus_call/src/scripts/compare_variant_calling_updated.py

```diff
@@ -0,0 +1,315 @@
+
+                    ## Add the formatted consensus CNV into a growing list of consensus CNVs
+                    consensus_list = consensus_list + [overlap_chrom]
+
+    return consensus_list
+
+
+
+def save_to_file(output_file_content, output_path, sample_name):
+    """ Function that takes in content, output file path, and sample name to save the content to a file1
+
+    Keyword arguments:
+    output_file_content -- content of consensus
+    output_path         -- path and name of the output file
+    sample_name         -- sample name of the consensus CNVs.
+    """
+
+    ## Open up the file to write in it, the 'w' option overwrites the file if the file exists.
+    with open( output_path, 'w') as file:
+
+        ## Add the sample name and file name to each line
+        single_name = os.path.basename(output_path)
+
+        ## Loop through the output file and print line by line
+        for k in output_file_content:
+
+            ## Join the sample_name and single_name(file name) to the CNV info
+            k.extend([sample_name, single_name,'\n'])
+            file.write('\t'.join(k))
+
+    sys.stderr.write('$$$ Write to file ' + str(output_path) + ' was sucessful\n')
+
+
+
+################### SCRIPT BODY #################################
+
+
+
+## Define parser for the input file
+parser = argparse.ArgumentParser(description="""This script takes in 3 bed files, each from one of the 3 callers
+                                                 and find common CNVs between two files at a time.""")
+parser.add_argument('--manta', required=True,
+                    help='path to the manta file')
+parser.add_argument('--cnvkit', required=True,
+                    help='path to the cnvkit file')
+parser.add_argument('--freec', required=True,
+                    help='path to the freec file')
+parser.add_argument('--manta_cnvkit', required=True,
+                    help='path of the output consensus between manta and cnvkit')
+parser.add_argument('--manta_freec', required=True,
+                    help='path of the output consensus between manta and freec')
+parser.add_argument('--cnvkit_freec', required=True,
+                    help='path of the output consensus between cnvkit and freec')
+parser.add_argument('--sample', default='no_sample',
+                    help='sample name to use in the file')
+
+args = parser.parse_args()
+
+
+## Read in the input files as dictionaries
+manta_dict = read_input_file(args.manta)
+cnvkit_dict = read_input_file(args.cnvkit)
+freec_dict = read_input_file(args.freec)
+
+## Put the input dictionaries into a bigger dictionary
+dict_of_input_content = {'manta':manta_dict, 'cnvkit':cnvkit_dict, 'freec':freec_dict}
+
+input_file_names = list(dict_of_input_content.keys())
+
+## Put the output file paths into their own lists that is to be iterated over
+## This order is important
+output_list = [args.manta_cnvkit, args.manta_freec, args.cnvkit_freec]
```

Comment:
Since the order is important so that file names line up with the combinations, lets move the caller pairs up, then use that to make sure the output list is in the same order we will generate later.

```suggestion
## Make a list for pairs of callers to make consensus for
caller_pairs = list(itertools.combinations(['manta', 'cnvkit', 'freec'],2))

# generate output list in the same order
output_files = [getattr(args, "_".join(callers) for callers in caller_pairs]
```